powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/v3/PowerAuthServiceImpl.java

```diff
@@ -205,7 +205,34 @@ public PrepareActivationResponse prepareActivation(PrepareActivationRequest requ
     @Override
     @Transactional
     public CreateActivationResponse createActivation(CreateActivationRequest request) throws GenericServiceException {
-        throw new IllegalStateException("Not implemented yet.");
+        try {
+            // Get request parameters
+            String userId = request.getUserId();
+            Date activationExpireTimestamp = XMLGregorianCalendarConverter.convertTo(request.getTimestampActivationExpire());
+            Long maxFailedCount = request.getMaxFailureCount();
```

Comment:
Could I double check where the default value (`5`) is set? In general, I believe we should do some basic request validation here, to make the future work on wultra/powerauth-server#9 a bit easier.

Reply:
Yes, this is handled as previously in the `initActivation()` method. By default the value is `null` and it is non-null only when it is modified.